clusterloader2/pkg/provider/provider.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"fmt"
+	"strings"
+)
+
+type providerCtor func(map[string]string) Provider
+
+var (
+	providerCtorMap map[string]providerCtor
+)
+
+func getProviderCtorMap() map[string]providerCtor {
+	if providerCtorMap == nil {
+		providerCtorMap = map[string]providerCtor{}
+	}
+	return providerCtorMap
+}
+
+// InitOptions encapsulates the fields needed to init provider.
+type InitOptions struct {
+	// TODO remove this and use providerConfigs.
```

Comment:
Avoid ownerless / contextless TODOs. Either create an issue an link it here, `TODO(#1234)` or assign it to you, `TODO(@Catramen): ` if you plan to do it yourself